### PR TITLE
solves issue #1280

### PR DESCRIPTION
**Pull request by sundaram2021**
traverseDFS() is not working properly ,

while logging it says  [ undefined ].

and there is also an issue with searchDFS() function,

the following issue i have faced

The stack array is being used as a queue instead of a stack. A queue follows the first-in, first-out (FIFO) principle, while a stack follows the last-in, first-out (LIFO) principle. For a DFS algorithm, we need to use a stack data structure.

The code is pushing elements to the stack using the tree[node.right] and tree[node.left] expressions, but it should be using node.right and node.left directly. The tree array should not be used here.

The code is using tree[0] as the starting node for the search, but it should be using the entire tree object as the starting node.

[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

### Describe your change:

- [ ] Add an algorithm?
- [x] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

### Checklist:

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new JavaScript files are placed inside an existing directory.
- [x] All filenames should use the UpperCamelCase (PascalCase) style. There should be no spaces in filenames.
      **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
- [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: Trees/DepthFirstSearch.js [hunk 1]
```diff
@@ -1,76 +1,213 @@
...
  * DFS Algorithm implementation in JavaScript
  * DFS Algorithm for traversing or searching graph data structures.
 */
 
-function traverseDFS (root) {
-  const stack = [root]
-  const res = []
+function dfs(tree) {
+  if (!tree) return;
+  console.log(tree.value);
```
Comment (appgurueu) [inline Trees/DepthFirstSearch.js:9]: This should either be calling a visitor function or yielding. No side effects.

File: Trees/DepthFirstSearch.js [hunk 2]
```diff
@@ -1,76 +1,213 @@
...
-      stack.push(curr.right)
-    }
+function searchDFS(tree, target) {
+  if (tree.value === target) {
+    return true;
+  }
 
-    if (curr.left) {
-      stack.push(curr.left)
+  for (let i = 0; i < tree.children.length; i++) {
```
Comment (appgurueu) [inline Trees/DepthFirstSearch.js:25]: Just use `for (const child of tree.children)`

File: Trees/DepthFirstSearch.js [hunk 3]
```diff
@@ -1,76 +1,213 @@
...
+// If it does, it returns true to indicate that the target value has been found. 
+// If not, the function loops over the children of the current node and calls itself recursively on each child. 
+// If any of the recursive calls returns true, the function also returns true to indicate that the target value 
+// has been found. If none of the recursive calls returns true, the function returns false to indicate that the 
+// target value was not found in the tree
 
-  while (stack.length !== 0) {
-    for (let i = 0; i < stack.length; i++) {
-      const node = stack.pop()
+const tree = {
```
Comment (appgurueu) [inline Trees/DepthFirstSearch.js:45]: This should be in a proper test case.

File: Trees/DepthFirstSearch.js [hunk 4]
```diff
@@ -1,76 +1,213 @@
...
     }
   }
 
-  return res.reverse()
+  return false;
 }
 
-function searchDFS (tree, value) {
-  const stack = []
+// This function takes two arguments: tree, which is the tree to be searched, and target,
```
Comment (appgurueu) [inline Trees/DepthFirstSearch.js:34]: This should be in a JS doc comment

File: Trees/DepthFirstSearch.js (hunk 1 above)
Comment (sundaram2021) [reply Trees/DepthFirstSearch.js:9]: But for what

File: Trees/DepthFirstSearch.js (hunk 2 above)
Comment (sundaram2021) [reply Trees/DepthFirstSearch.js:25]: yeah we can use it , its a matter of preference but here it provides much more fine control

File: Trees/DepthFirstSearch.js (hunk 1 above)
Comment (appgurueu) [reply Trees/DepthFirstSearch.js:9]: For reusability.

File: Trees/DepthFirstSearch.js (hunk 2 above)
Comment (appgurueu) [reply Trees/DepthFirstSearch.js:25]: I don't see why the index would be needed.

Comment (sundaram2021) [reply Trees/DepthFirstSearch.js]: I have done that

File: Trees/DepthFirstSearch.js (hunk 3 above)
Comment (sundaram2021) [reply Trees/DepthFirstSearch.js:45]: can you provide some more info on that  like what kind of test you want,

it will help me to make changes then....

Comment (appgurueu) [reply Trees/DepthFirstSearch.js:45]: See https://github.com/TheAlgorithms/TypeScript/blob/master/CONTRIBUTING.md#writing-good-tests

**Comment by appgurueu** [comment]
Issue was closed via another PR.

**Pull request closed without merging**